### PR TITLE
Add GHA workflow to build and test RPMs

### PR DESCRIPTION
**Pull request by brianjmurrell**

### CODE REVIEW

File: .github/workflows/rpm-build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+name: RPM Build and Test
+
+env:
+  # TODO: Surely this is a GH variable of some sort for the name of the repo
+  NAME: argobots
```
Comment (daltonbohning) [inline .github/workflows/rpm-build-and-test.yml:5]: Not tested but some googling says this should work
```
"${{github.repository}}" | cut -d "/" -f 2
```

And this, but NOT for PRs unfortunately
```
${{ github.event.repository.name }}
```

File: .github/workflows/rpm-build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      rpm-test-version:
+        description: 'RPM version to test'
+        required: false
+  pull_request:
+  push:
+    branches:
+      - master
+
+concurrency:
+  group: rpm-build-and-test-${{ github.head_ref  || github.run_id }}
```
Comment (daltonbohning) [inline .github/workflows/rpm-build-and-test.yml:25]: I think you can use this variable for the workflow
```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

**Review by daltonbohning — APPROVED** [review]

File: .github/workflows/rpm-build-and-test.yml (hunk 2 above)
Comment (brianjmurrell) [reply .github/workflows/rpm-build-and-test.yml:25]: Ah, yes.  Another nice one I will push if I don't feel the squeeze of time closing in on me.

File: .github/workflows/rpm-build-and-test.yml (hunk 1 above)
Comment (brianjmurrell) [reply .github/workflows/rpm-build-and-test.yml:5]: I knew there had to be one but just feeling the time is running out to get mired in the small details.

Great suggestion though.  If I feel like the time will allow, I'll push an update for that.

Comment (daltonbohning) [reply .github/workflows/rpm-build-and-test.yml:5]: Agree. No need to stress on this

File: .github/workflows/rpm-build-and-test.yml (hunk 2 above)
Comment (daltonbohning) [reply .github/workflows/rpm-build-and-test.yml:25]: Agree

File: .github/workflows/rpm-build-and-test.yml (hunk 1 above)
Comment (brianjmurrell) [reply .github/workflows/rpm-build-and-test.yml:5]: #28 is this PR with your suggestions applied.  It seems to have done the right thing but is just getting started.  If #28 completes testing before my time runs out, we can land it but if it doesn't this one should be landable very soon.

**Comment by brianjmurrell** [comment]
Landed in 5f45767.

**Pull request closed without merging**